### DISCUSSION TITLE
New Custom Templates

### DISCUSSION THREAD

**Discussion by rxerium** [projectdiscovery/nuclei#4414] [Q&A] [answered]
Hi there,

I would like to request the ability to add custom made templates (that are not in the `nuclei-templates` repo) to a scan without the need of maintaining a local folder containing my templates. Currently for my workflows I use any templates with the `kev` tag in addition to my custom templates I have created, all of which are managed locally in a folder I maintain manually. 

It would be useful if there is a way to use templates both from the nuclei-templates repo in addition to custom templates security engineers create. I had the idea of pointing to a personal GitHub repo with my custom templates with the `-custom-template` flag which would look something like:
`nuclei -custom-template https://gituhb.com..../..... -tags kev -etags tech -l input.txt -o output.txt`
This may not be the best idea and more than happy for the mechanics to be decided by the PD team or other security researchers in the community. 

This will be really useful especially if security researchers would like to use templates that are not accepted in the nuclei-templates repo (passive version detection templates for example). I also see this become more of a feature request given that more templates are being created after the release of templates.nuclei.sh. 

Do let me know your thoughts, I think this will be a useful addition to the tool.

Many thanks,

Rishi

**Comment by tarunKoyalwar** [comment, marked as answer]
@rxerium nuclei already supports using custom templates from various sources including github , gitlab , azure and more here's more docs on that https://docs.projectdiscovery.io/tools/nuclei/running#custom-templates
